stacker/util.py

```diff
@@ -505,9 +505,65 @@ def ensure_s3_bucket(s3_client, bucket_name, bucket_region):
             raise
 
 
+class Extractor(object):  # pylint: disable=too-few-public-methods
+    """Base class for extractors."""
+
+    def __init__(self, archive=None):
+        """
+        Setup up extractor object with the archive path.
+
+        Args:
+            archive (string): Archive path
+        """
+        self.archive = archive
+
+
+class TarExtractor(Extractor):
+    """Extracts tar archives."""
+
+    def extract(self, destination):
+        """Extract the archive."""
+        with tarfile.open(self.archive, 'r:') as tar:
+            tar.extractall(path=destination)
+
+    @staticmethod
+    def extension():
+        """Return archive extension."""
+        return '.tar'
+
+
+class TarGzipExtractor(Extractor):
+    """Extracts compressed tar archives."""
+
+    def extract(self, destination):
+        """Extract the archive."""
+        with tarfile.open(self.archive, 'r:gz') as tar:
+            tar.extractall(path=destination)
+
+    @staticmethod
+    def extension():
+        """Return archive extension."""
+        return '.tar.gz'
+
+
+class ZipExtractor(Extractor):
+    """Extracts zip archives."""
+
+    def extract(self, destination):
+        """Extract the archive."""
+        with zipfile.ZipFile(self.archive, 'r') as zip_ref:
+            zip_ref.extractall(destination)
+
+    @staticmethod
+    def extension():
+        """Return archive extension."""
+        return '.zip'
+
+
 class SourceProcessor():
     """Makes remote python package sources available in the running python
        environment."""
+    CACHE_DATE_FORMAT = '%Y-%m-%d-%H-%M-%S'
```

Comment:
Hmm, any issues with time zones here?

Reply:
Also, can we go with a standard like iso8601 if we're not going to use a unix timestamp?